src/nodebuilder.cpp

```diff
@@ -69,11 +71,38 @@ void NodeBuilder::OnMapStart(const Mark& mark, const std::string& tag,
   node.set_tag(tag);
   node.set_style(style);
   m_mapDepth++;
+  m_mergeDicts.emplace_back();
+}
+
+void MergeMapCollection(detail::node& map_to, detail::node& map_from,
+                        detail::shared_memory_holder& pMemory) {
```

Comment:
```suggestion
static void MergeMapCollection(detail::node& map_to, detail::node&& map_from,
                        detail::shared_memory_holder& pMemory) {
```
This function should be `static` or in an anonymous namespace to not leak any symbols to other translation units.

I also believe this is also a good case for a r-value reference for `map_from`. So I would use a `&&` to make clear that `map_from` will be in an unclear state after this call.

Reply:
>This function should be static or in an anonymous namespace to not leak any symbols to other translation units.

Done

>I also believe this is also a good case for a r-value reference for map_from. So I would use a && to make clear that map_from will be in an unclear state after this call.

I'm not sure about this one, since `map_from` is in valid state upon return from `MergeMapCollection`, what make you thinks the opposite?

Reply:
I don't know what I was thinking last month....It seems fine way it is :sweat_smile: